lib/graphql/batch/executor.rb

```diff
@@ -35,5 +36,21 @@ def wait_all
     def clear
       loaders.clear
     end
+
+    def defer
```

Comment:
promise.rb defines it to just yield so that it can be overridden by a derived class to actually defer callbacks.  E.g. you could defer the callback to run on next tick when using EventMachine.  I'm just using this hook to set `loading` to `false` in the callback.  I would prefer not to defer calling the callback though, since that results in backtraces that are missing the context to debug exceptions.